File: shanti_education/public/js/purchase_invoice.js
frappe.ui.form.on("Purchase Invoice", {
    onload: function(frm) {
        // Ignore cancellation for all linked documents of respective DocTypes.
        frm.ignore_doctypes_on_cancel_all = ["Payment Ledger Entry"];
    },
    cost_center: function(frm) {
        for (let row of frm.doc.taxes) {
            frappe.model.set_value(row.doctype, row.name, "cost_center", frm.doc.cost_center);
        }
        for (let row of frm.doc.items) {
            frappe.model.set_value(row.doctype, row.name, "cost_center", frm.doc.cost_center);
        }
    },
});

File: shanti_education/shanti_education/doc_event/purchase_invoice.py
from frappe import _


def before_validate(self, method):
    update_cost_centre_in_items(self)

def update_cost_centre_in_items(self):
    for item in self.items:
        item.cost_center = self.cost_center
    for tax in self.taxes:
        tax.cost_center = self.cost_center